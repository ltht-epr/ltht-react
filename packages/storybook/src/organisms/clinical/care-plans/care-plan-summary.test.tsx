import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import CarePlanSummary from '@ltht-react/care-plan-summary'
import carePlans from './care-plan.fixtures'

const Sut: React.FC = () => {
  return (
    <>
      {carePlans.map(carePlan => (
        <div key={carePlan.id}>
          <CarePlanSummary carePlan={carePlan} />
        </div>
      ))}
    </>
  )
}

describe('CarePlanSummary', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Sut />, div)
  })
  it('matches snapshot', () => {
    expect(mount(<Sut />)).toMatchSnapshot('wrapper mount')
  })
})
