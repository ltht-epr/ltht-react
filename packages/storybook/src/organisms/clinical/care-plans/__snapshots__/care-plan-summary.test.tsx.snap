// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarePlanSummary matches snapshot: wrapper mount 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: left;
}

.emotion-0 {
  color: #768692;
}

.emotion-3 {
  text-align: right;
}

.emotion-2 {
  color: #231f20;
}

.emotion-7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-5 {
  color: #231f20;
  text-align: left;
}

.emotion-6 {
  color: #768692;
  text-align: left;
  font-size: smaller;
  padding-top: 0.25rem;
}

.emotion-10 {
  text-align: right;
}

.emotion-9 {
  color: #768692;
  text-align: right;
  font-size: smaller;
  padding-top: 0.25rem;
}

<Sut>
  <div
    key="580ca927-34e0-e911-a2c7-005056926fe4|careplan"
  >
    <CarePlanSummary
      carePlan={
        Object {
          "author": Array [
            Object {
              "display": "Dr Joe Bloggs",
              "typeName": "Test",
            },
            Object {
              "display": "Dr Adam Careplan",
              "typeName": "Test",
            },
          ],
          "careTeam": Array [
            Object {
              "display": "LYPFT Care Team 1",
              "typeName": "Team",
            },
          ],
          "description": "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo",
          "id": "580ca927-34e0-e911-a2c7-005056926fe4|careplan",
          "intent": "PLAN",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": true,
            "requestedWhen": "",
          },
          "period": Object {
            "end": Object {
              "kind": "DATE",
              "value": "2014-02-02T00:00:00+00:00",
            },
            "start": Object {
              "kind": "DATE",
              "value": "2013-01-01T00:00:00+00:00",
            },
          },
          "status": "ACTIVE",
          "text": Object {
            "div": "",
            "text": "Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.",
          },
          "title": "Mental Health Crisis Plan",
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <CarePlanRedacted
            carePlan={
              Object {
                "author": Array [
                  Object {
                    "display": "Dr Joe Bloggs",
                    "typeName": "Test",
                  },
                  Object {
                    "display": "Dr Adam Careplan",
                    "typeName": "Test",
                  },
                ],
                "careTeam": Array [
                  Object {
                    "display": "LYPFT Care Team 1",
                    "typeName": "Team",
                  },
                ],
                "description": "Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo",
                "id": "580ca927-34e0-e911-a2c7-005056926fe4|careplan",
                "intent": "PLAN",
                "metadata": Object {
                  "dataSources": Array [
                    Object {
                      "display": "Mock",
                    },
                  ],
                  "isRedacted": true,
                  "requestedWhen": "",
                },
                "period": Object {
                  "end": Object {
                    "kind": "DATE",
                    "value": "2014-02-02T00:00:00+00:00",
                  },
                  "start": Object {
                    "kind": "DATE",
                    "value": "2013-01-01T00:00:00+00:00",
                  },
                },
                "status": "ACTIVE",
                "text": Object {
                  "div": "",
                  "text": "Itaque earum rerum hic tenetur a sapiente delectus, ut aut reiciendis voluptatibus maiores alias consequatur aut perferendis doloribus asperiores repellat.",
                },
                "title": "Mental Health Crisis Plan",
              }
            }
          >
            <div
              className="emotion-1"
            >
              <RedactedDescription>
                <div
                  className="emotion-0"
                >
                  Insufficient privileges
                </div>
              </RedactedDescription>
            </div>
            <div
              className="emotion-3"
            >
              <PeriodSummary
                period={
                  Object {
                    "end": Object {
                      "kind": "DATE",
                      "value": "2014-02-02T00:00:00+00:00",
                    },
                    "start": Object {
                      "kind": "DATE",
                      "value": "2013-01-01T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Jan-01,-2013 to Feb-02,-2014
                </div>
              </PeriodSummary>
            </div>
          </CarePlanRedacted>
        </div>
      </Styled(div)>
    </CarePlanSummary>
  </div>
  <div
    key="690ca927-68e0-c511-b2c7-898056926ze4|careplan"
  >
    <CarePlanSummary
      carePlan={
        Object {
          "author": Array [
            Object {
              "display": "John Mclane",
              "typeName": "Test",
            },
          ],
          "careTeam": Array [
            Object {
              "display": "LYPFT Care Team 2",
              "typeName": "Team",
            },
          ],
          "id": "690ca927-68e0-c511-b2c7-898056926ze4|careplan",
          "intent": "PROPOSAL",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "period": Object {
            "end": Object {
              "kind": "DATE",
              "value": "2015-03-04T00:00:00+00:00",
            },
            "start": Object {
              "kind": "DATE",
              "value": "2014-02-02T00:00:00+00:00",
            },
          },
          "status": "COMPLETED",
          "title": "Test Care Plan",
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <Styled(div)>
            <div
              className="emotion-7"
            >
              <CarePlanTitle
                carePlan={
                  Object {
                    "author": Array [
                      Object {
                        "display": "John Mclane",
                        "typeName": "Test",
                      },
                    ],
                    "careTeam": Array [
                      Object {
                        "display": "LYPFT Care Team 2",
                        "typeName": "Team",
                      },
                    ],
                    "id": "690ca927-68e0-c511-b2c7-898056926ze4|careplan",
                    "intent": "PROPOSAL",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2015-03-04T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                    },
                    "status": "COMPLETED",
                    "title": "Test Care Plan",
                  }
                }
              >
                <div
                  className="emotion-5"
                >
                  Test Care Plan
                </div>
              </CarePlanTitle>
              <CarePlanDescription
                carePlan={
                  Object {
                    "author": Array [
                      Object {
                        "display": "John Mclane",
                        "typeName": "Test",
                      },
                    ],
                    "careTeam": Array [
                      Object {
                        "display": "LYPFT Care Team 2",
                        "typeName": "Team",
                      },
                    ],
                    "id": "690ca927-68e0-c511-b2c7-898056926ze4|careplan",
                    "intent": "PROPOSAL",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2015-03-04T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                    },
                    "status": "COMPLETED",
                    "title": "Test Care Plan",
                  }
                }
              >
                <div
                  className="emotion-6"
                >
                  John Mclane - LYPFT Care Team 2
                </div>
              </CarePlanDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-10"
            >
              <PeriodSummary
                period={
                  Object {
                    "end": Object {
                      "kind": "DATE",
                      "value": "2015-03-04T00:00:00+00:00",
                    },
                    "start": Object {
                      "kind": "DATE",
                      "value": "2014-02-02T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Feb-02,-2014 to Mar-04,-2015
                </div>
              </PeriodSummary>
              <CarePlanStatus
                carePlan={
                  Object {
                    "author": Array [
                      Object {
                        "display": "John Mclane",
                        "typeName": "Test",
                      },
                    ],
                    "careTeam": Array [
                      Object {
                        "display": "LYPFT Care Team 2",
                        "typeName": "Team",
                      },
                    ],
                    "id": "690ca927-68e0-c511-b2c7-898056926ze4|careplan",
                    "intent": "PROPOSAL",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2015-03-04T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                    },
                    "status": "COMPLETED",
                    "title": "Test Care Plan",
                  }
                }
              >
                <div
                  className="emotion-9"
                >
                  Proposal - Completed
                </div>
              </CarePlanStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </CarePlanSummary>
  </div>
  <div
    key="gpvisit"
  >
    <CarePlanSummary
      carePlan={
        Object {
          "activity": Object {
            "detail": Object {
              "performer": Array [
                Object {
                  "display": "Nurse Nancy",
                  "reference": "Practitioner/13",
                  "typeName": "",
                },
              ],
              "status": "COMPLETED",
            },
            "outcomeReference": Array [
              Object {
                "display": "",
                "reference": "Encounter/example",
                "typeName": "",
              },
            ],
          },
          "addresses": Array [
            Object {
              "display": "obesity",
              "reference": "#p1",
              "typeName": "",
            },
          ],
          "author": Array [
            Object {
              "display": "Dr Joe Bloggs",
              "typeName": "Test",
            },
            Object {
              "display": "Dr Adam Careplan",
              "typeName": "Test",
            },
          ],
          "careTeam": Array [
            Object {
              "display": "",
              "reference": "#careteam",
              "typeName": "",
            },
          ],
          "goal": Array [
            Object {
              "display": "",
              "reference": "#goal",
              "typeName": "",
            },
          ],
          "id": "gpvisit",
          "intent": "PLAN",
          "metadata": Object {
            "dataSources": Array [
              Object {
                "display": "Mock",
              },
            ],
            "isRedacted": false,
            "requestedWhen": "",
          },
          "period": Object {
            "start": Object {
              "kind": "DATE",
              "value": "2014-02-02T00:00:00+00:00",
            },
          },
          "status": "ACTIVE",
          "text": Object {
            "div": "<div xmlns=\\"http://www.w3.org/1999/xhtml\\">
      <p>  Represents the flow of a patient within a practice. The plan is created when
        they arrive and represents the 'care' of the patient over the course of that encounter.
        They first see the nurse for basic observations (BP, pulse, temp) then the doctor for
        the consultation and finally the nurse again for a tetanus immunization. As the plan is
        updated (e.g. a new activity added), different versions of the plan exist, and workflow timings
        for reporting can be gained by examining the plan history. This example is the version after
        seeing the doctor, and waiting for the nurse.The plan can either be created 'ad hoc' and modified as
        the parient progresses, or start with a standard template (which can, of course, be altered to suit the patient.</p>
    </div>",
            "status": "ADDITIONAL",
            "text": "",
          },
          "title": "Test Care Plan",
        }
      }
    >
      <Styled(div)>
        <div
          className="emotion-4"
        >
          <Styled(div)>
            <div
              className="emotion-7"
            >
              <CarePlanTitle
                carePlan={
                  Object {
                    "activity": Object {
                      "detail": Object {
                        "performer": Array [
                          Object {
                            "display": "Nurse Nancy",
                            "reference": "Practitioner/13",
                            "typeName": "",
                          },
                        ],
                        "status": "COMPLETED",
                      },
                      "outcomeReference": Array [
                        Object {
                          "display": "",
                          "reference": "Encounter/example",
                          "typeName": "",
                        },
                      ],
                    },
                    "addresses": Array [
                      Object {
                        "display": "obesity",
                        "reference": "#p1",
                        "typeName": "",
                      },
                    ],
                    "author": Array [
                      Object {
                        "display": "Dr Joe Bloggs",
                        "typeName": "Test",
                      },
                      Object {
                        "display": "Dr Adam Careplan",
                        "typeName": "Test",
                      },
                    ],
                    "careTeam": Array [
                      Object {
                        "display": "",
                        "reference": "#careteam",
                        "typeName": "",
                      },
                    ],
                    "goal": Array [
                      Object {
                        "display": "",
                        "reference": "#goal",
                        "typeName": "",
                      },
                    ],
                    "id": "gpvisit",
                    "intent": "PLAN",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                    },
                    "status": "ACTIVE",
                    "text": Object {
                      "div": "<div xmlns=\\"http://www.w3.org/1999/xhtml\\">
      <p>  Represents the flow of a patient within a practice. The plan is created when
        they arrive and represents the 'care' of the patient over the course of that encounter.
        They first see the nurse for basic observations (BP, pulse, temp) then the doctor for
        the consultation and finally the nurse again for a tetanus immunization. As the plan is
        updated (e.g. a new activity added), different versions of the plan exist, and workflow timings
        for reporting can be gained by examining the plan history. This example is the version after
        seeing the doctor, and waiting for the nurse.The plan can either be created 'ad hoc' and modified as
        the parient progresses, or start with a standard template (which can, of course, be altered to suit the patient.</p>
    </div>",
                      "status": "ADDITIONAL",
                      "text": "",
                    },
                    "title": "Test Care Plan",
                  }
                }
              >
                <div
                  className="emotion-5"
                >
                  Test Care Plan
                </div>
              </CarePlanTitle>
              <CarePlanDescription
                carePlan={
                  Object {
                    "activity": Object {
                      "detail": Object {
                        "performer": Array [
                          Object {
                            "display": "Nurse Nancy",
                            "reference": "Practitioner/13",
                            "typeName": "",
                          },
                        ],
                        "status": "COMPLETED",
                      },
                      "outcomeReference": Array [
                        Object {
                          "display": "",
                          "reference": "Encounter/example",
                          "typeName": "",
                        },
                      ],
                    },
                    "addresses": Array [
                      Object {
                        "display": "obesity",
                        "reference": "#p1",
                        "typeName": "",
                      },
                    ],
                    "author": Array [
                      Object {
                        "display": "Dr Joe Bloggs",
                        "typeName": "Test",
                      },
                      Object {
                        "display": "Dr Adam Careplan",
                        "typeName": "Test",
                      },
                    ],
                    "careTeam": Array [
                      Object {
                        "display": "",
                        "reference": "#careteam",
                        "typeName": "",
                      },
                    ],
                    "goal": Array [
                      Object {
                        "display": "",
                        "reference": "#goal",
                        "typeName": "",
                      },
                    ],
                    "id": "gpvisit",
                    "intent": "PLAN",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                    },
                    "status": "ACTIVE",
                    "text": Object {
                      "div": "<div xmlns=\\"http://www.w3.org/1999/xhtml\\">
      <p>  Represents the flow of a patient within a practice. The plan is created when
        they arrive and represents the 'care' of the patient over the course of that encounter.
        They first see the nurse for basic observations (BP, pulse, temp) then the doctor for
        the consultation and finally the nurse again for a tetanus immunization. As the plan is
        updated (e.g. a new activity added), different versions of the plan exist, and workflow timings
        for reporting can be gained by examining the plan history. This example is the version after
        seeing the doctor, and waiting for the nurse.The plan can either be created 'ad hoc' and modified as
        the parient progresses, or start with a standard template (which can, of course, be altered to suit the patient.</p>
    </div>",
                      "status": "ADDITIONAL",
                      "text": "",
                    },
                    "title": "Test Care Plan",
                  }
                }
              >
                <div
                  className="emotion-6"
                >
                  Dr Joe Bloggs, Dr Adam Careplan
                </div>
              </CarePlanDescription>
            </div>
          </Styled(div)>
          <Styled(div)>
            <div
              className="emotion-10"
            >
              <PeriodSummary
                period={
                  Object {
                    "start": Object {
                      "kind": "DATE",
                      "value": "2014-02-02T00:00:00+00:00",
                    },
                  }
                }
              >
                <div
                  className="emotion-2"
                >
                  Feb-02,-2014 to Present
                </div>
              </PeriodSummary>
              <CarePlanStatus
                carePlan={
                  Object {
                    "activity": Object {
                      "detail": Object {
                        "performer": Array [
                          Object {
                            "display": "Nurse Nancy",
                            "reference": "Practitioner/13",
                            "typeName": "",
                          },
                        ],
                        "status": "COMPLETED",
                      },
                      "outcomeReference": Array [
                        Object {
                          "display": "",
                          "reference": "Encounter/example",
                          "typeName": "",
                        },
                      ],
                    },
                    "addresses": Array [
                      Object {
                        "display": "obesity",
                        "reference": "#p1",
                        "typeName": "",
                      },
                    ],
                    "author": Array [
                      Object {
                        "display": "Dr Joe Bloggs",
                        "typeName": "Test",
                      },
                      Object {
                        "display": "Dr Adam Careplan",
                        "typeName": "Test",
                      },
                    ],
                    "careTeam": Array [
                      Object {
                        "display": "",
                        "reference": "#careteam",
                        "typeName": "",
                      },
                    ],
                    "goal": Array [
                      Object {
                        "display": "",
                        "reference": "#goal",
                        "typeName": "",
                      },
                    ],
                    "id": "gpvisit",
                    "intent": "PLAN",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                      "isRedacted": false,
                      "requestedWhen": "",
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                    },
                    "status": "ACTIVE",
                    "text": Object {
                      "div": "<div xmlns=\\"http://www.w3.org/1999/xhtml\\">
      <p>  Represents the flow of a patient within a practice. The plan is created when
        they arrive and represents the 'care' of the patient over the course of that encounter.
        They first see the nurse for basic observations (BP, pulse, temp) then the doctor for
        the consultation and finally the nurse again for a tetanus immunization. As the plan is
        updated (e.g. a new activity added), different versions of the plan exist, and workflow timings
        for reporting can be gained by examining the plan history. This example is the version after
        seeing the doctor, and waiting for the nurse.The plan can either be created 'ad hoc' and modified as
        the parient progresses, or start with a standard template (which can, of course, be altered to suit the patient.</p>
    </div>",
                      "status": "ADDITIONAL",
                      "text": "",
                    },
                    "title": "Test Care Plan",
                  }
                }
              >
                <div
                  className="emotion-9"
                >
                  Plan - Active
                </div>
              </CarePlanStatus>
            </div>
          </Styled(div)>
        </div>
      </Styled(div)>
    </CarePlanSummary>
  </div>
</Sut>
`;
